backon/src/lib.rs

```diff
@@ -114,13 +118,15 @@ pub use sleep::Sleeper;
 #[cfg(all(not(target_arch = "wasm32"), feature = "tokio-sleep"))]
 pub use sleep::TokioSleeper;
 
+#[cfg(feature = "std")]
 mod blocking_retry;
-pub use blocking_retry::BlockingRetry;
-pub use blocking_retry::BlockingRetryable;
+#[cfg(feature = "std")]
```

Comment:
It doesn't make sense to me to hide retry under `std`. The motivation here is that we can't use `std::thread::sleep` with `no_std`, correct? I believe the proper way to address this is to implement something like `BlockingSleep` and only provide `DefaultSleeper` under `std`.

Reply:
didn't want to go down implementing for different platforms, not sure the best way to go.
is there a specific platform in mind? think it might be best to hide it behind std for now
the default implementation is std so feature parity would still be the same but blocking might be tricky

Reply:
Let me give a try first.